domenetjenester/medlem/src/main/java/no/nav/foreldrepenger/domene/medlem/MedlemTjeneste.java

```diff
@@ -79,23 +84,33 @@ public EndringsresultatSnapshot finnAktivGrunnlagId(Long behandlingId) {
             return Optional.empty();
         }
         var behandlingsresultat = behandlingsresultatOpt.get();
-        var medlemskapsvilkåret = finnVilkår(behandlingsresultat, VilkårType.MEDLEMSKAPSVILKÅRET);
-
-        if (medlemskapsvilkåret.isPresent()) {
-            var medlem = medlemskapsvilkåret.get();
-            if (medlem.getGjeldendeVilkårUtfall().equals(VilkårUtfallType.OPPFYLT)) {
-                var løpendeOppfylt = finnVilkår(behandlingsresultat, VilkårType.MEDLEMSKAPSVILKÅRET_LØPENDE)
-                    .filter(v -> v.getGjeldendeVilkårUtfall().equals(VilkårUtfallType.OPPFYLT));
-                if (løpendeOppfylt.isEmpty()) {
-                    var behandling = behandlingRepository.hentBehandling(behandlingId);
-                    return medlemskapVilkårPeriodeRepository.hentOpphørsdatoHvisEksisterer(behandling);
-                }
+        var medlemskapsvilkåret = behandlingsresultat.getVilkårResultat()
+            .getVilkårene()
+            .stream()
+            .filter(vilkår -> vilkår.getVilkårType().equals(VilkårType.MEDLEMSKAPSVILKÅRET))
+            .filter(vilkår -> VilkårUtfallType.erFastsatt(vilkår.getGjeldendeVilkårUtfall()))
+            .findFirst();
+        if (medlemskapsvilkåret.isEmpty()) {
+            return Optional.empty();
```

Comment:
Forutgående - eller satse på at det ikke er noen rader?

Reply:
Tenker at opphørsdato ikke er relevant for behandlingene med forutgående